ERCS/erc-7715.md

```diff
@@ -0,0 +1,466 @@
+
+[ERC-7679](./eip-7679.md) UserOp Builder contract defines `bytes calldata context` parameter in all of its methods. It’s equivalent to the`permissionsContext` returned by the `wallet_issuePermissions` call.
+
+Example of formatting userOp signature using the [ERC-7679](./eip-7679.md) UserOp Builder
+
+```jsx
+const getSignature = async ({ address, userOp, permissionsContext }: GetSignatureArgs) => {
+    return readContract(config, {
+      abi: BUILDER_CONTRACT_ABI,
+      address: BUILDER_CONTRACT_ADDRESS,
+      functionName: 'getSignature',
+      args: [address, userOp, permissionsContext]
+    })
+  }
+```
+
+**Example of the entire flow:**
+
+```mermaid
+sequenceDiagram
+    participant D as DApp
+    participant W as Wallet
+    actor U as User
+    participant SA as SmartAccount
+
+    D ->>+ W: wallet_issuePermissions
+    W ->>+ W: Build the permission object
+    W ->>+ U: Show Permission UI
+    U ->>+ W: Approve and sign the permissions object <br> to grant permissions
+    W ->>+ D: Return wallet_issuePermissions response
+    D ->>+ D: Store permissionsContext
+    Note left of D: DApp has permissions context <br> and is looking to send UserOp 
+    D ->>+ D: Build UserOp and Sign with Session Key
+    alt Leverage ERC-7679 UserOpBuilder
+    D ->>+ SA: getNonce (with permissionsContext)
+    D ->>+ D: update UserOp with nonce
+    D ->>+ SA: getCallData (with permissionsContext)
+    D ->>+ D: update UserOp with callData
+    D ->>+ SA: getSignature (with permissionsContext)
+    D ->>+ D: update UserOp with signature
+    end
+    D ->>+ D: Send UserOperation
+```
+
+#### ERC-7710
+
+When requesting permissions with a `type` of `account`, the returned data will be redeemable using the interfaces specified in ERC-7710. This allows the recipient of the permissions to use any account type (EOA or contract) to form a transaction or UserOp using whatever payment or relay infrastructure they prefer, by sending an internal message to the returned `permissions.signerData.managerAddress` and calling its `function redeemDelegation(bytes calldata _data, Action calldata _action) external;` function with the `_data` parameter set to the returned `permissions.permissionsContext`, and the `_action` data forming the message that the permissions recipient desires the user's account to emit, as defined by this struct:
+
+```
+struct Action {
+    address to;
+    uint256 value;
+    bytes data;
+}
+```
+
+A simple pseudocode example of using a permission in this way, given two ethers signers in the same context, where `alice` wants to request a permission from `bob` might be like this:
+
+```
+// Alice requests a permission from Bob
+const permissionsResponse = await bob.request({
+  method: 'wallet_issuePermissions',
+  params: [{
+    signer: {
+      type: 'account',
+      data: {
+        id: bob.address
```

Comment:
Earlier you describe this field as
> `signer` is an optional field that identifies the key or account associated with the session

And here Alice requests a session from Bob. Wouldn't the session's signer then be 

```suggestion
        id: alice.address
```